exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/AbstractElasticsearchExporterIntegrationTest.java

```diff
@@ -0,0 +1,181 @@
+import org.elasticsearch.action.admin.indices.settings.get.GetSettingsRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.settings.Settings;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+
+public abstract class AbstractElasticsearchExporterIntegrationTest {
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  protected final ExporterIntegrationRule exporterBrokerRule = new ExporterIntegrationRule();
+
+  protected ElasticsearchNode<ElasticsearchForkedJvm> elastic;
+  protected ElasticsearchExporterConfiguration configuration;
+  protected ElasticsearchExporterFaultToleranceIT.ElasticsearchTestClient esClient;
+
+  @Before
+  public void setUp() {
+    elastic = new ElasticsearchForkedJvm(temporaryFolder);
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    if (esClient != null) {
+      esClient.close();
+      esClient = null;
+    }
+
+    exporterBrokerRule.stop();
+    elastic.stop();
+    configuration = null;
+  }
+
+  protected void assertIndexSettings() {
+    final ImmutableOpenMap<String, Settings> settingsForIndices = esClient.getSettingsForIndices();
+    for (ObjectCursor<String> key : settingsForIndices.keys()) {
+      final Settings settings = settingsForIndices.get(key.value);
+      final Integer numberOfShards = settings.getAsInt("index.number_of_shards", -1);
+      final Integer numberOfReplicas = settings.getAsInt("index.number_of_replicas", -1);
+
+      assertThat(numberOfShards)
+          .withFailMessage(
+              "Expected number of shards of index %s to be 1 but was %d", key.value, numberOfShards)
+          .isEqualTo(1);
+      assertThat(numberOfReplicas)
+          .withFailMessage(
+              "Expected number of replicas of index %s to be 0 but was %d",
+              key.value, numberOfReplicas)
+          .isEqualTo(0);
+    }
+  }
+
+  protected void assertRecordExported(Record<?> record) {
+    final Map<String, Object> source = esClient.get(record);
+    assertThat(source)
+        .withFailMessage("Failed to fetch record %s from elasticsearch", record)
+        .isNotNull();
+
+    assertThat(source).isEqualTo(recordToMap(record));
+  }
+
+  protected ElasticsearchTestClient createElasticsearchClient(
+      ElasticsearchExporterConfiguration configuration) {
+    return new ElasticsearchTestClient(
+        configuration, new ZbLogger("io.zeebe.exporter.elasticsearch"));
+  }
+
+  protected Map<String, Object> recordToMap(final Record<?> record) {
```

Comment:
Didn't we have these kind of helper somewhere else?

Reply:
Maybe? It's marked as an addition, but really I just moved it to the new abstract IT class. You mean somewhere else in the code base?

Reply:
Ah no seems there exist only the to msgpack helpers